core/src/main/java/com/netflix/conductor/core/utils/Utils.java

```diff
@@ -119,10 +111,26 @@ public static void checkNotNullOrEmpty(String input, String errorMessage) {
      * @throws ApplicationException if input object is not valid
      */
     public static void checkNotNull(Object object, String errorMessage) {
-        try {
-            Preconditions.checkNotNull(object, errorMessage);
-        } catch (NullPointerException exception) {
-            throw new ApplicationException(ApplicationException.Code.INVALID_INPUT, errorMessage);
+        if (object == null) {
+            throw new IllegalArgumentException(errorMessage);
```

Comment:
`NullPointerException`

Reply:
fixed

Reply:
Shouldn't this be `NullPointerException`?